src/v/iceberg/values_json.cc

```diff
@@ -0,0 +1,699 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "iceberg/values_json.h"
+
+#include "iceberg/json_utils.h"
+
+#include <absl/strings/numbers.h>
+#include <absl/strings/str_split.h>
+#include <boost/iostreams/device/array.hpp>
+#include <boost/iostreams/stream.hpp>
+#include <boost/range/irange.hpp>
+
+#include <charconv>
+#include <chrono>
+#include <iomanip>
+#include <ranges>
+#include <stdexcept>
+
+namespace {
+
+using namespace iceberg;
+
+std::string_view
+parse_string_view(const json::Value& data, std::string_view context = "") {
+    if (!data.IsString()) {
+        throw std::invalid_argument(fmt::format(
+          "Expected JSON string for {}, got {}", context, data.GetType()));
+    }
+    return std::string_view{data.GetString(), data.GetStringLength()};
+}
+
+iobuf hex_str_to_iobuf(std::string_view str) {
```

Comment:
absl::HexStringToBytes could simplify this method.

Reply:
it seems that the good version of this function (that validates the input string) isn't available in the cmake build, so I don't think using it really simplifies things in practice (compared to streaming)

Reply:
😞

Reply:
for the beautiful moment I was building w/ bazel only, the code did look much better 🙃